src/@types/config.d.ts

```diff
@@ -0,0 +1,342 @@
+// TODO: This was auto generated from defaultConfig. Needs to be verified.
+
+import DOMPurify from 'dompurify';
+
+export interface MermaidConfig {
+  theme?: string;
+  themeVariables?: any;
+  themeCSS?: string;
+  maxTextSize?: number;
+  darkMode?: boolean;
+  fontFamily?: string;
+  logLevel?: number;
+  securityLevel?: string;
+  startOnLoad?: boolean;
+  arrowMarkerAbsolute?: boolean;
+  secure?: string[];
+  deterministicIds?: boolean;
+  deterministicIDSeed?: string;
+  flowchart?: FlowchartDiagramConfig;
+  sequence?: SequenceDiagramConfig;
+  gantt?: GanttDiagramConfig;
+  journey?: JourneyDiagramConfig;
+  class?: ClassDiagramConfig;
+  state?: StateDiagramConfig;
+  er?: ErDiagramConfig;
+  pie?: PieDiagramConfig;
+  requirement?: RequirementDiagramConfig;
+  gitGraph?: GitGraphDiagramConfig;
+  c4?: C4DiagramConfig;
+  dompurifyConfig?: DOMPurify.Config;
+}
+
+// TODO: More configs needs to be moved in here
+export interface BaseDiagramConfig extends BaseDiagramConfig {
+  useWidth?: number;
+  useMaxWidth?: boolean;
+}
```

Comment:
@knsv we should move all the common configs that diagrams share here.

Reply:
Agree

Reply:
Can you help figure out what the eligible configs are?